### PR TITLE
Add name-based (version 5) UUIDs generator, since it uses SHA-1 over MD5

### PR DESCRIPTION
**Pull request by ayurchuk**
Add name-based UUIDv5 generator that supports SHA-1. This is preferable way of generating UUID comparing to default one that uses MD5 algorithm.
Also unused UUID_EX pattern from UUIDs class was removed.

@jypma, @alar17 please review.

### CODE REVIEW

**Comment by jypma** [comment]
LGTM 👍

**Comment by jypma** [comment]
actually @ayurchuk could you add a few sanity unit tests, e.g. verifying that the same strings yield the same target UUID? Just to exercise the code a bit.

**Comment by ayurchuk** [comment]
@jypma makes sense, added basic one. I changed version to SNAPSHOT here.

File: ts-reaktive-java/src/test/java/com/tradeshift/reaktive/NameBasedUUIDSpec.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+@RunWith(CuppaRunner.class)
+public class NameBasedUUIDSpec {{
+
+    describe("NameBasedUUID", () -> {
+        final UUID namespace = UUID.fromString("49da330a-35f6-40f2-8e33-570e79859030");
+        final UUID actorId = UUID.fromString("e330948c-efd7-4cae-b6cc-1c4118d47649");
+        final String clientToken = "7f1b889b-ebc2-409b-89db-f7ec7932ddef";
+
+        it("should return one UUID for the same input parameters", () -> {
+            assertEquals(
```
Comment (jypma) [inline ts-reaktive-java/src/test/java/com/tradeshift/reaktive/NameBasedUUIDSpec.java]: we use [AssertJ](http://joel-costigliola.github.io/assertj/) in this project. So 
```
assertThat(
  NameBasedUUID.create(namespace, actorId + ":" + clientToken)
).isEqualTo(
  NameBasedUUID.create(namespace, actorId + ":" + clientToken))
);
```
etc.

Comment (ayurchuk) [inline ts-reaktive-java/src/test/java/com/tradeshift/reaktive/NameBasedUUIDSpec.java]: cool! just moved all assertions to AssertJ.

**Pull request merged**